### PR TITLE
Fix unmarshalling for additionalProperties

### PR DESCRIPTION
**Pull request by catalinaperalta**
Unmarshalling for BlobMetadata additional properties was being skipped during unmarshalling because no custom unmarshaller was being generated for the BlobMetadata struct. 

Make sure key and value pairs are both included in the map. 
 
Fix this case for all additional properties.

### CODE REVIEW

File: src/generator/xmlAdditionalProps.ts [hunk 1]
```diff
@@ -26,22 +26,31 @@ type additionalProperties map[string]*string
 // UnmarshalXML implements the xml.Unmarshaler interface for additionalProperties.
 func (ap *additionalProperties) UnmarshalXML(d *xml.Decoder, start xml.StartElement) error {
```
Comment (jhendrixMSFT) [inline src/generator/xmlAdditionalProps.ts:27]: We should adopt the implementation from track 1.5 (unsure why I didn't do this to begin with).

https://github.com/Azure/azure-storage-blob-go/blob/master/azblob/zz_generated_models.go#L38

File: src/generator/models.ts [hunk 2]
```diff
@@ -793,6 +793,9 @@ function generateAliasType(structDef: StructDef, receiver: string, forMarshal: b
     if (prop.schema.type === SchemaType.DateTime) {
       text += `\t\t${prop.language.go!.name} *${prop.schema.language.go!.internalTimeType} \`${structDef.Language.marshallingFormat}:"${sn}"\`\n`;
     } else if (prop.language.go!.isAdditionalProperties || prop.language.go!.needsXMLDictionaryUnmarshalling) {
+      if (prop.language.go!.isAdditionalProperties && prop.language.go!.needsXMLDictionaryUnmarshalling) {
+        sn = ',any'
```
Comment (jhendrixMSFT) [inline src/generator/models.ts:797]: Why is this needed?

File: src/generator/xmlAdditionalProps.ts (hunk 1 above)
Comment (catalinaperalta) [reply src/generator/xmlAdditionalProps.ts:27]: I did take a look at the implementation for track 1.5 it doesnt unmarshal the content for BlobMetadata properly. The XML could be malformed though. Since it for the additional properties the startElement is empty for some reason.

**Comment by jhendrixMSFT** [comment]
Closing as this is an issue with the swagger and will be fixed at that level.
